source/common/tracing/http_tracer_impl.cc

```diff
@@ -244,11 +245,38 @@ LightStepDriver::LightStepDriver(const Json::Object& config,
   });
 }
 
-SpanPtr LightStepDriver::startSpan(const std::string& operation_name, SystemTime start_time) {
-  lightstep::Span span = tls_.getTyped<TlsLightStepTracer>(tls_slot_).tracer_.StartSpan(
-      operation_name, {lightstep::StartTimestamp(start_time)});
+SpanPtr LightStepDriver::startSpan(Http::HeaderMap& request_headers,
+                                   const std::string& operation_name, SystemTime start_time) {
+  lightstep::Tracer& tracer = tls_.getTyped<TlsLightStepTracer>(tls_slot_).tracer_;
+  LightStepSpanPtr active_span;
+
+  // Extract downstream context from HTTP carrier.
+  const std::string parent_context =
```

Comment:
storing to variable here is not needed, just check if header exists, then pass c_str() directly to ParseFromString. (In the future it would be nice to avoid string conversion).